ThePalaceProject/library-registry-admin
# Upgrade to Node 18.

## Description

Upgrades this project to build with Node 18, along with some other quality/security improvements. This includes:

- Upgrading dependencies that were incompatible with Node 18
- Upgrading dependencies to resolve npm audit security errors (not all of them, just the easy ones for now)
- Code/configuration changes required by major version upgrades of dependencies, including
   - Major webpack update required updating some webpack config files and removing some no-longer-supported arguments
   - Major mocha (test runner) update revealed some tests that weren't actually running, and were failing when they started running
- Replacing the opds-web-client dependency with the Palace fork (@thepalaceproject/web-opds-client)

## Motivation and Context

The project had been using Node 12, which is no longer supported, and will not be allowed in GitHub actions soon. Notion: https://www.notion.so/lyrasis/Upgrade-library-registry-admin-to-Node-18-5540aa350641472782118f508c172b54

## How Has This Been Tested?

- Verify that `npm install` runs without error
- Verify that all npm scripts in package.json run without error, especially `npm test`.
- Verify that the app runs in dev:
  - `npm run dev-server -- --env=backend=https://registry-tpp-qa.palaceproject.io`
  - Open http://localhost:8080/admin in a browser
  - Log in using qa registry credentials
  - Verify that all UI functionality works

## Checklist:
